pipelines/library/src/common/VersionInfo.groovy

```diff
@@ -148,53 +150,66 @@ class VersionInfo {
-                "(?:jdk\\-)?(?<version>${vnumRegex}(\\-${preRegex})?\\+${buildRegex}(\\-${optRegex})?)".toString(),
-                "(?:jdk\\-)?(?<version>${vnumRegex}\\-${preRegex}(\\-${optRegex})?)".toString(),
-                "(?:jdk\\-)?(?<version>${vnumRegex}(\\+\\-${optRegex})?)".toString()
-        ]
-
-        for (String regex : version223Regexs) {
-            context.println "[INFO] Attempting to match223 regex: ${regex}"
-            final matched223 = versionString =~ /^${regex}.*/
-            if (matched223.matches()) {
-                context.println "[SUCCESS] match223 regex matched!"
-
-                major = or0(matched223, 'major')
-                minor = or0(matched223, 'minor')
-                security = or0(matched223, 'security')
-
-                if (matched223.group('pre') != null) {
-                    pre = matched223.group('pre')
-                    context.println "[SUCCESS] regex group (pre) of last 223 regex matched to ${pre}"
-                }
+        String regex = "(?:jdk\\-)?(?<version>${vnumRegex}(\\-${preRegex})?\\+${buildRegex}(\\-${optRegex})?)".toString()
+        context.println "[INFO] Attempting to match223 regex: ${regex}"
+        final matched223 = versionString =~ /^${regex}.*/
+        if (matched223.matches()) {
+            context.println "[SUCCESS] match223 regex matched!"
 
-                build = or0(matched223, 'build')
+            major = or0(matched223, 'major')
+            minor = or0(matched223, 'minor')
+            security = or0(matched223, 'security')
 
-                if (matched223.group('opt') != null) {
-                    opt = matched223.group('opt')
-                    context.println "[SUCCESS] regex group (opt) of last 223 regex matched to ${opt}"
-                }
+            // Match optional .fix
+            if (matched223.group('fix') != null) {
+                fix = matched223.group('fix') as Integer
+                context.println "[SUCCESS] regex group (fix) of 223 regex matched to ${fix}"
+            }
+
+            if (matched223.group('pre') != null) {
+                pre = matched223.group('pre')
+                context.println "[SUCCESS] regex group (pre) of 223 regex matched to ${pre}"
+            }
+
+            build = or0(matched223, 'build')
 
-                version = matched223.group('version')
-                return true
+            if (matched223.group('opt') != null) {
+                opt = matched223.group('opt')
+                context.println "[SUCCESS] regex group (opt) of 223 regex matched to ${opt}"
             }
-            context.println "[WARNING] Failed to match last 223 regex."
+
+            version = matched223.group('version')
+            return true
         }
+        context.println "[WARNING] Failed to match 223 regex."
 
         return false
     }
 
     String formSemver() {
         if (major != null) {
             def semver = major + "." + minor + "." + security
+ 
+            if (fix) {
```

Comment:
not sure about this, according to the semver standard you can only have 3 numbers, not sure we can do this and still call it a semver